pisa/core/binning.py

```diff
@@ -956,6 +971,18 @@ def bin_widths(self):
             self._bin_widths = np.abs(np.diff(self.bin_edges.m)) * self.units
         return self._bin_widths
 
+    @property
+    def weighted_bin_widths(self):
+        """Absolute widths of bins."""
+        if self._weighted_bin_widths is None:
+            if self.is_log:
+                self._weighted_bin_widths = (
+                    self.edge_magnitudes[1:] / self.edge_magnitudes[:-1]
```

Comment:
I think we have to take the log of this to make the sums of two binnings add up again!

Reply:
After testing my stage again I can confirm that taking the log of the ratio here produces the correct total event rates after upsampling. If we make this log, I would suggest also placing a comment in the code telling people why it has to be this way so that no one accidentally breaks the upsampling stage.

Reply:
I can add a comment, but a comment _and_ a unit test would be even better! Can you checkout from my branch here, make the modifications you propose, add an appropriate unit test that catches this issue (so we have more than just the comment in the code to check us -- though I think that's a good idea, too!), and submit a PR against my branch? I can merge there, then merge the whole thing into PISA. (I've got a lot going on right now, so you doing this will help move things along faster.)

Reply:
Ok yes, I can do that, but for this evening I think I have coded myself out... I can get to it tomorrow!